src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -762,17 +798,64 @@ export class SearchSource {
     return field;
   }
 
+  public async loadDataViewFields() {
+    const dataView = this.getField('index');
+    if (!dataView || !(dataView instanceof DataViewLazy)) {
+      return;
+    }
+    // eslint-disable-next-line no-console
+    console.log('loadDataViewFields because of DataViewLazy');
+    const request = this.mergeProps(this, { body: {} }, ['query', 'filter']);
+    let fields = dataView.timeFieldName ? [dataView.timeFieldName] : [];
+    const sort = this.getField('sort');
+    if (sort) {
+      const sortArr = Array.isArray(sort) ? sort : [sort];
+      for (const s of sortArr) {
+        const keys = Object.keys(s);
+        fields = fields.concat(keys);
+      }
+    }
+    for (const query of request.query) {
+      if (query.query) {
+        const nodes = fromKueryExpression(query.query);
+        const queryFields = getKueryFields([nodes]);
+        fields = fields.concat(queryFields);
+      }
+    }
+    const filters = request.filters;
+    if (filters) {
+      const filtersArr = Array.isArray(filters) ? filters : [filters];
+      for (const f of filtersArr) {
+        fields = fields.concat(f.meta.field);
+      }
+    }
+    fields = fields.filter((f) => Boolean(f));
+
+    if (dataView.getSourceFiltering() && dataView.getSourceFiltering().excludes.length) {
+      // if source filtering is enabled, we need to fetch all the fields
```

Comment:
How was this arrived at? I just don't see why its needed based on existing search source code but I might be overlooking something.

Reply:
I don't know exactly where we do this in search source, didn't look, but for sure it needs to be done

then there is data view source filtering enabled, we need to send all other fields that we don't need to filter out ... using the fields API (default), works differently in using `_source`.  You could configure a data view with a field excluded, add a debugger in the search source code, and watch what and where it happens in the code 

however, the solution is pretty simple. if there are fields to be filtered, fetch all fields. here we can't work around until ES supports excluding fields in this request